main.go

```diff
@@ -53,12 +53,14 @@ func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var serviceNameTemplate string
+	var allowServiceName bool
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 	flag.StringVar(&serviceNameTemplate, "default-backendservice-name", "{name}-{port}",
 		"A naming template consists of {namespace}, {name}, {port} or {hash} separated by hyphens, "+
 			"where {hash} is the first 8 digits of a hash of other given information")
+	flag.BoolVar(&allowServiceName, "enable-supplied-service-names", true, "Enable using custom service names in autoneg annotation.")
```

Comment:
I like how you call them "custom service names" in the description. Can we make the flag name match, i.e. `enable-custom-service-names` here and in the documentation?

Otherwise, looks great!